helix-core/src/main/java/org/apache/helix/HelixManagerProperty.java

```diff
@@ -66,11 +66,8 @@ private HelixManagerProperty(String version, long healthReportLatency,
   }
 
   public HelixCloudProperty getHelixCloudProperty() {
-    return _helixCloudProperty;
-  }
-
-  public void setHelixCloudProperty(CloudConfig cloudConfig) {
-    _helixCloudProperty = new HelixCloudProperty(cloudConfig);
+    return _helixCloudProperty == null ? HelixCloudProperty.EMPTY_HELIX_CLOUD_PROPERTY
+        : _helixCloudProperty;
```

Comment:
Can we pre-populate the default value so that `_helixCloudProperty` is always not null?

Reply:
Thanks for the suggestion, but pre-populate default value won't help here because this variable is assigned in `HelixManagerProperty` constructor, not in a setter. So if null is passed into the constructor, even with pre-populated value, it will still be assigned to null.

Reply:
Thanks for the info.
`it will still be assigned to null.` This is technically right, but is also the confusing part. Does a null value make sense or even be accessible at all? The public getter returns either the non-null assigned value, or an empty new instance. If possible, let's try to do the null check at assignment time. Also please let me know if I misunderstood anything.

Reply:
Thanks for explaining. I was trying to prevent a potential NPE so I return a new instance instead of null, but practically the old code doesn't have functionality concerns.
Do a null check at assignment time is technically correct but is not necessary because:
1. There are multiple places (two constructors and a setter function) that this value is set, so this can lead to duplicated code.
2. There are Helix managers that will not be related to cloud at all, so this field is not useful. If we assign default value to it, it can cause memory waste.
I modified my change to do lazy instantiation on this field, but kept it in the getter. Hope you find this reasonable. Thanks.

Reply:
That makes sense, thanks.